mteb/models/model2vec_models.py

```diff
@@ -0,0 +1,209 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import Sequence
+from functools import partial
+from typing import Any
+
+import numpy as np
+
+from mteb.encoder_interface import PromptType
+from mteb.model_meta import ModelMeta
+
+from .wrapper import Wrapper
+
+logger = logging.getLogger(__name__)
+
+
+class Model2VecWrapper(Wrapper):
+    def __init__(
+        self,
+        model_name: str,
+        **kwargs,
+    ) -> None:
+        """Wrapper for Model2Vec models.
+
+        Args:
+            model: The Model2Vec model to use. Can be a string (model name), a SentenceTransformer model, or a CrossEncoder model.
+        """
+        try:
+            from model2vec import StaticModel
+        except ModuleNotFoundError as e:
+            raise ModuleNotFoundError(
+                "To use the Model2Vec models `model2vec` is required. Please install it with `pip install mteb[model2vec]`."
+            ) from e
+
+        self.model_name = model_name
+        self.static_model = StaticModel.from_pretrained(self.model)
+
+    def encode(
+        self,
+        sentences: Sequence[str],
+        **kwargs: Any,
+    ) -> np.ndarray:
+        """Encodes the given sentences using the encoder.
+
+        Args:
+            sentences: The sentences to encode.
+            **kwargs: Additional arguments to pass to the encoder.
+
+        Returns:
+            The encoded sentences.
+        """
+        return self.static_model.encode(sentences)
+
+
+m2v_base_glove_subword = ModelMeta(
+    loader=partial(
+        Model2VecWrapper,
+        model_name="minishlab/M2V_base_glove_subword",
+    ),
+    name="minishlab/M2V_base_glove_subword",
+    languages=["eng_Latn"],
+    open_weights=True,
+    revision="5f4f5ca159b7321a8b39739bba0794fa0debddf4",
+    release_date="2024-10-14",  # blog post date
+    n_parameters=103 * 1e6,
+    max_tokens=1000000,  # I took this from the current leaderboard, I'm not sure if it's unlimited in theory, tough
+    embed_dim=256,
+    license="mit",
+    similarity_fn_name="cosine",
+    framework=["PyTorch"],
+    reference="https://huggingface.co/minishlab/M2V_base_glove_subword",
+    use_instructions=False,
+    adapted_from="BAAI/bge-base-en-v1.5",
+    superseded_by=None,
+)
+
+
+m2v_base_glove = ModelMeta(
+    loader=partial(
+        Model2VecWrapper,
+        model_name="minishlab/M2V_base_glove",
+    ),
+    name="minishlab/M2V_base_glove",
+    languages=["eng_Latn"],
+    open_weights=True,
+    revision="38ebd7f10f71e67fa8db898290f92b82e9cfff2b",
+    release_date="2024-10-14",  # blog post date
```

Comment:
For the M2V models the release date is 2024-09-21 (minor detail though, feel free to ignore).